File: src/main/java/br/org/acal/resouces/model/WaterQualityModel.java
package br.org.acal.resouces.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.time.LocalDate;

import static jakarta.persistence.GenerationType.IDENTITY;

@Entity
@Table(name = "parametro_coleta")
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class WaterQualityModel implements Serializable {

    @Id
    @Column(name = "ide_parametro_coleta")
    @GeneratedValue(strategy = IDENTITY)
    private String number;

    @Basic(optional = false)
    @Column(name = "ide_tipo_parametro")
    private String param;

    @Basic(optional = false)
    @Column(name = "exigido")
    private String required;

    @Basic(optional = false)
    @Column(name = "analisado")
    private String analyzed;

    @Basic(optional = false)
    @Column(name = "conformidade")
    private String accordance;

    @Basic(optional = false)
    @Column(name = "data")
    private LocalDate date;

}

File: src/main/kotlin/br/org/acal/resources/model/WaterQualityModel.kt
package br.org.acal.resources.model

import jakarta.persistence.Basic
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.io.Serializable
import java.time.LocalDate
import lombok.AllArgsConstructor
import lombok.Builder
import lombok.Data
import lombok.NoArgsConstructor

@Entity
@Table(name = "parametro_coleta")
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
class WaterQualityModel : Serializable {
    @Id
    @Column(name = "ide_parametro_coleta")
    private var number: String? = null

    @Basic(optional = false)
    @Column(name = "ide_tipo_parametro")
    private var param: String? = null

    @Basic(optional = false)
    @Column(name = "exigido")
    private var required: String? = null

    @Basic(optional = false)
    @Column(name = "analisado")
    private var analyzed: String? = null

    @Basic(optional = false)
    @Column(name = "conformidade")
    private var accordance: String? = null

    @Basic(optional = false)
    @Column(name = "data")
    private var date: LocalDate? = null
}